staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -119,6 +120,26 @@ type ValidatingAdmissionPolicySpec struct {
 	// Allowed values are Ignore or Fail. Defaults to Fail.
 	// +optional
 	FailurePolicy *FailurePolicyType `json:"failurePolicy,omitempty" protobuf:"bytes,4,opt,name=failurePolicy,casttype=FailurePolicyType"`
+
+	// TODO align this with the webhook PR: https://github.com/kubernetes/kubernetes/pull/116261
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	MatchConditions []MatchCondition `json:"matchConditions" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,5,rep,name=matchConditions"`
+}
+
+var _ cel.ExpressionAccessor = &MatchCondition{}
+
+// TODO align this with the Webhook PR: https://github.com/kubernetes/kubernetes/pull/116261
+type MatchCondition struct {
+	Name       string `json:"name,required" protobuf:"bytes,5,rep,name=name"`
+	Expression string `json:"expression,required protobuf:" protobuf:"bytes,6,opt,name=expression"`
+}
+
+func (m *MatchCondition) GetExpression() string {
```

Comment:
This should not be in this file?

Reply:
Should it not? This is what allows us to feed v1alpha1 MatchCondition directly into the controller (which is how the controller is currently written for the rest of these types).

Not having GetExpression() would mean converting it into some intermediate type, which is doable, but out of alignment with the rest of the code.

Which is preferred?

Reply:
Implementing `ExpressionAccessor` by adding `GetExpression()` is the right things to do. (@cici37 this was added in @ivelichkovich's refactor)

Reply:
We generally not prefer to have impl related details in types.go from API perspective. files under `staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy` might be a better place for this :)

Reply:
We could have a separately file for this like what messageExpr did or matcher.go also looks good(I am not sure if interface.go would fit though...)
Ref: https://github.com/kubernetes/kubernetes/blob/7488bf5861776b428d1f1da2f8986ca6ae60cc51/staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/message.go

Reply:
Used the type in the matchconditions package.